satpy/writers/cf_writer.py

```diff
@@ -499,14 +514,20 @@ def da2cf(dataarray, epoch=EPOCH, flatten_attrs=False, exclude_attrs=None, compr
                 If True, flatten dict-type attributes
             exclude_attrs (list):
                 List of dataset attributes to be excluded
-
+            valid_cf_dataset_name (bool):
+                Prepend dataset name with CHANNEL_ if staring with a digit
         """
         if exclude_attrs is None:
             exclude_attrs = []
 
         new_data = dataarray.copy()
         if 'name' in new_data.attrs:
             name = new_data.attrs.pop('name')
+            if valid_cf_dataset_name and name[0].isdigit():
+                _orig_name = name
+                name = 'CHANNEL_' + name
+                warnings.warn('Rename dataset {} to {}.'.format(_orig_name, name))
+                new_data.attrs['satpy_dataset_name'] = _orig_name
```

Comment:
I suppose this will be included in the resulting netcdf files, do we really want this?

Reply:
Yes it will be included.

If we use static `CHANNEL_` we could strip of that when reading. But if you let the user select what string to use, it is impossible to know.